test/puppetlabs/kitchensink/classpath_test.clj

```diff
@@ -1,12 +1,18 @@
 (ns puppetlabs.kitchensink.classpath-test
   (:require [clojure.test :refer :all]
-            [puppetlabs.kitchensink.classpath :refer [with-additional-classpath-entries]])
+            [dynapath.dynamic-classpath :refer [classpath-urls]]
+            [puppetlabs.kitchensink.classpath :refer [add-classpath
+                                                      with-additional-classpath-entries]])
   (:import (java.net URL)))
 
 (deftest with-additional-classpath-entries-test
   (let [paths ["/foo" "/bar"]
         get-urls #(into #{}
-                        (.getURLs (.getContextClassLoader (Thread/currentThread))))]
+                        (classpath-urls (.getContextClassLoader (Thread/currentThread))))]
+    ;; Called for side effect of ensuring there's a modifiable classloader
+    ;; without needing to make that function public
+    (add-classpath "file:/nil")
```

Comment:
This part seems weird. How does `add-classpath` get called otherwise in real use?

Reply:
It gets called on its own in trapperkeeper. It does not normally need to be called before calling the next line, but this test wants to use some url methods after with-additional-classpath-entries removes the temporary classloader it puts there, so this makes that possible. It's not entirely clean.

Reply:
the way this test works doesn't make that much sense to begin with

Reply:
cleaned this up so it's not confusing anymore